rust/kernel/user_ptr.rs

```diff
@@ -242,6 +244,27 @@ impl UserSlicePtrWriter {
         self.len() == 0
     }
 
+    /// Writes zeroes to the user slice.
+    pub fn clear(&mut self, mut len: usize) -> KernelResult {
+        let mut ret = Ok(());
+        if len > self.1 {
+            ret = Err(Error::EFAULT);
+            len = self.1;
+        }
```

Comment:
I think the documentation should explain the behavior better, i.e. that it will clear no matter what, even if the length is bigger and that the fields will be updated in all cases.

In any case, isn't a bit confusing to do that given `write_raw()` will return without performing any action if `len` is out of bounds? (Which, by the way, should likely be a `EINVAL` rather than a `EFAULT`, no?)

Reply:
> I think the documentation should explain the behavior better, i.e. that it will clear no matter what, even if the length is bigger and that the fields will be updated in all cases.

Sure. I updated the documentation.

> In any case, isn't a bit confusing to do that given `write_raw()` will return without performing any action if `len` is out of bounds? 

Since `write_raw` doesn't update its internal state to reflect partial writes, it not so problematic at the moment.

For `clear` we need to report how much was actually cleared because we want to mimic the behaviour of the C implementation.

I agree that it isn't ideal, but I think the solution here is really to update `write_raw` to reflect what was written in its internal state. But since this PR is about `clear`, I think this should be updated in another PR.

> (Which, by the way, should likely be a `EINVAL` rather than a `EFAULT`, no?)

Perhaps. But I think `EINVAL`, if bubbled up to userspace, is too generic. `EFAULT` is a better indication that the problem is in some address. Also, this is the existing behaviour, I'm just trying to add `clear`.

Reply:
Thanks! It is clear now.